### PR TITLE
feat: Mask wrapped as MaskInverse

### PR DESCRIPTION
**Pull request by elvizlai**
fix: #4

### CODE REVIEW

**Comment by elvizlai** [comment]
@mennanov PTAL

File: copy_proto_test.go [hunk 1]
```diff
@@ -191,6 +191,37 @@ func TestStructToStruct_MaskInverse(t *testing.T) {
...
+		Permissions: []testproto.Permission{testproto.Permission_EXECUTE},
+		Name:        &testproto.User_FemaleName{FemaleName: "Dana"},
+		Friends: []*testproto.User{
+			{Id: 2, Username: "friend1"},
+			{Id: 3, Username: "friend2"},
+		},
+	}
+	userDst := &testproto.User{}
+	// Mask to MaskInverse
+	mask := fieldmask_utils.MaskInverse(fieldmask_utils.Mask{"Id": fieldmask_utils.Mask{}, "Friends": fieldmask_utils.Mask{"Username": fieldmask_utils.Mask{}}})
```
Comment (mennanov) [inline copy_proto_test.go]: I think the right way to do that is to create a new helper function `MaskInverseFromMask(Mask) MaskInverse` that will create a `MaskInverse` from a given `Mask` recursively (including all the submasks)

Comment (elvizlai) [reply copy_proto_test.go]: Maybe no need to create one.
for MaskInverse: subFilter.IsEmpty() means the sub should be filtered.

Comment (mennanov) [reply copy_proto_test.go]: I've added some tests, PTAL https://github.com/mennanov/fieldmask-utils/commit/83cb71b9bb683e1d733faa67c292cf392f43b483

Is that what you're looking for?

Comment (elvizlai) [reply copy_proto_test.go]: yes, nil and empty should be same strategy when convert from Mask to MakeInverse.

Comment (mennanov) [inline copy_proto_test.go]: This is equivalent to 
```
mask := fieldmask_utils.MaskInverse{"Id": fieldmask_utils.Mask{}, "Friends": fieldmask_utils.Mask{"Username": fieldmask_utils.Mask{}}}
```
If so, it is expected that `Id` is _not_ copied and only `Friends.Username` are copied.
So the test case is invalid. 
Please, take a look at https://github.com/mennanov/fieldmask-utils/pull/6

Comment (elvizlai) [reply copy_proto_test.go]: sorry for mistake, pr #6 should revert for bad test case.
@mennanov  PTAL

**Comment by elvizlai** [comment]
See #8 , Close this

**Pull request closed without merging**